tfx/components/example_gen/utils.py

```diff
@@ -324,9 +353,60 @@ def _find_matched_span_version_from_path(
           matched_version_int)
 
 
+def _get_spec_width(spec_full_regex: Text, spec_name: Text,
+                    split: example_gen_pb2.Input.Split) -> Optional[Text]:
+  """Returns width modifier of a spec, if it exists."""
+  result = re.search(spec_full_regex, split.pattern)
+  assert result, 'No %s found in split %s' % (spec_name, split.pattern)
+  width_str = result.group('width')
+  if width_str:
+    try:
+      width_int = int(width_str)
+      if width_int <= 0:
+        raise ValueError('Not a positive integer.')
+    except ValueError:
+      raise ValueError(
+          'Width modifier in %s spec is not a positive integer: %s' %
+          (spec_name, split.pattern))
+  return width_str
+
+
+def _get_span_replace_glob_and_regex(
+    range_config: range_config_pb2.RangeConfig,
+    is_match_span: bool, is_match_date: bool, span_width_str: Optional[Text]
+) -> Union[Text, List[Text]]:
+  """Replace span or date spec if static range RangeConfig is provided."""
+  if range_config.HasField('static_range'):
+    if is_match_span:
+      # If using RangeConfig.static_range, replace span spec in patterns
+      # with given span from static range.
+      span_str = str(range_config.static_range.start_span_number)
+      if span_width_str:
+        span_width_int = int(span_width_str)
+        if span_width_int < len(span_str):
+          raise ValueError(
+              'Span spec width is less than number of digits in span: (%s, %s)'
+              % (span_width_int, span_str))
+        span_str = span_str.zfill(span_width_int)
+      return span_str
+
+    elif is_match_date:
+      # If using RangeConfig.static_range, replace date specs in patterns
+      # with calendar date derived from given span from static range.
+      span_int = range_config.static_range.start_span_number
+      year, month, day = span_number_to_date(span_int)
+      date_tokens = [str(year).zfill(4), str(month).zfill(2), str(day).zfill(2)]
+      return date_tokens
+
+  else:
+    raise ValueError("Only static_range in RangeConfig is supported.")
+
+
 def _create_matching_glob_and_regex(
     uri: Text, split: example_gen_pb2.Input.Split, is_match_span: bool,
-    is_match_date: bool, is_match_version: bool) -> Tuple[Text, Text]:
+    is_match_date: bool, is_match_version: bool,
+    range_config: Optional[range_config_pb2.RangeConfig]
+) -> Tuple[Text, Text]:
```

Comment:
given this function is quite complex, consider to elaborate the docstring for future readers.